internal/support/notifications/normal_distribution.go

```diff
@@ -16,10 +16,13 @@
 package notifications
 
 import (
+	"github.com/edgexfoundry/edgex-go/internal/support/notifications/interfaces"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 )
 
-func distributeAndMark(n models.Notification) error {
+type NormalDistributor struct{}
```

Comment:
Could we possibly remove this type, since it does not encapsulate any data and it's only method can be expressed as an exported helper function? I am thinking that we could make this `NormalDistributor` type an executor by doing the following:

1. Move this to the `notification/operator` directory
1. Update the name to match the other executors, maybe something like `DistributionExecutor`
1. Update the name of the `distributeAndMark` method to `Execute` the signature looks good as is.
1. Update the executor to use a "loader" or "updater" type rather than the `dbClient` directly.
1. Create an exported constructor

All the above steps are so that our operators are consistent.

I think this approach also will allow us to eliminate the need for the `distributor` global variable in the init.go file : 
```
var distributor interfaces.Distributor
```

and instead we can just create the executor when we need it and call the `Execute` method like we do for other operators in our REST handler functions. Overall, I do not think we need a global variable here.

Reply:
Agree. I created the global var out of need in order to mock and test notification add executor. 

Moving this to its own executor sounds good. However, when I started implementing this I realised that code in normal_distribution ultimately has calls into sending_service.go and escalation_service.go. I could possibly put all of those functions in one operator, however, it seems to me like they should be in several and, of course, ease of testing needs to be considered also. I'd like to hear your opinion on this